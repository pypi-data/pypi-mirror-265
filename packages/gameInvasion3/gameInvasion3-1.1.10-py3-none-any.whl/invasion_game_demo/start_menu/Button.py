import pygame

class Button:
    def __init__(self, rect, color, text, font, text_color, action=None):
        self.rect = rect
        self.color = color
        self.text = text
        self.font = font
        self.text_color = text_color
        self.action = action
        self.clicked = False

    def draw(self, screen):
        pygame.draw.rect(screen, self.color, self.rect)
        self.draw_text(screen)

    def draw_text(self, screen):
        text_surface = self.font.render(self.text, True, self.text_color)
        text_rect = text_surface.get_rect(center=self.rect.center)
        screen.blit(text_surface, text_rect)

    def check_click(self, event):
        if event.type == pygame.MOUSEBUTTONDOWN and event.button == 1:
            if self.rect.collidepoint(event.pos):
                self.clicked = True  # 设置为 True，表示按钮被点击
                if self.action is not None:
                    self.action()

    def get_click_status(self):
        return self.clicked

