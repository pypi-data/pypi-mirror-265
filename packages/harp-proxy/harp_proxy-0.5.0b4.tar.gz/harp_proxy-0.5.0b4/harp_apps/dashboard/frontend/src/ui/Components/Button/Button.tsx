import { StyledComponent } from "@emotion/styled"
import { HTMLAttributes, ReactNode } from "react"
import tw, { styled } from "twin.macro"

type ButtonVariant = "primary" | "secondary"

interface ButtonProps {
  children?: ReactNode
  variant?: ButtonVariant
}

const StyledComponents: Record<ButtonVariant, StyledComponent<JSX.IntrinsicElements["button"]>> = {
  primary: styled.button(() => [
    tw`px-2.5 py-1.5`,
    tw`text-sm font-semibold`,
    tw`text-white bg-blue-600 hover:bg-blue-500 focus-visible:outline-blue-500`,
    tw`focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2`,
    tw`ring-1 ring-inset ring-blue-700 `,
  ]),
  secondary: styled.button(() => [
    tw`px-2.5 py-1.5`,
    tw`text-sm font-semibold`,
    tw`bg-white text-gray-900 hover:bg-gray-50`,
    tw`ring-1 ring-inset ring-gray-300 `,
  ]),
}

const Button = ({ children, variant = "primary", ...props }: ButtonProps & HTMLAttributes<HTMLButtonElement>) => {
  const Wrapper = StyledComponents[variant]
  return <Wrapper {...props}>{children}</Wrapper>
}

export { Button }
