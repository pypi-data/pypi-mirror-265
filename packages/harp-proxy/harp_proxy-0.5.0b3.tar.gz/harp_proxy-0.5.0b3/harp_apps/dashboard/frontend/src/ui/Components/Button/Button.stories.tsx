import { Button } from "./Button"

export const Default = () => (
  <>
    <Button variant="secondary">Secondary</Button>
    <Button variant="primary">Primary</Button>
  </>
)
export const Primary = () => <Button>Hello</Button>
export const Secondary = () => <Button variant="secondary">Hello</Button>
