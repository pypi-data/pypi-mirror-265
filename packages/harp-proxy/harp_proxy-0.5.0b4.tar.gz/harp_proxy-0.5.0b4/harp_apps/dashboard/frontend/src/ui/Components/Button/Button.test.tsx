import { render, screen } from "@testing-library/react"
import { expect, describe, it } from "vitest"

import { Button } from "./Button"

describe("Button", () => {
  it("renders its children correctly", () => {
    render(<Button>Hello, world!</Button>)
    expect(screen.getByText("Hello, world!")).toBeInTheDocument()
  })

  it("forwards additional props to the underlying button element", () => {
    render(<Button data-testid="my-button">Test</Button>)
    expect(screen.getByTestId("my-button")).toBeInTheDocument()
  })

  it("renders correctly", () => {
    const { container } = render(<Button>Test Button</Button>)
    expect(container).toMatchSnapshot()
  })
})
