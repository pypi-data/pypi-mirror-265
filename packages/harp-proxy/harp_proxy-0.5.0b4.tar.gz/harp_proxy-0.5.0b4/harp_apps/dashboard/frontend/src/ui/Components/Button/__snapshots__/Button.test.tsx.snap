// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > renders correctly 1`] = `
<div>
  <button
    class="css-1fo8x6t"
  >
    Test Button
  </button>
</div>
`;
